sysboundary/sysboundarycondition.cpp

```diff
@@ -878,4 +916,40 @@ namespace SBC {
    
    /*! Get a bool telling whether to call again applyInitialState upon restarting the simulation. */
    bool SysBoundaryCondition::doApplyUponRestart() const {return this->applyUponRestart;}
+
+   bool OuterBoundaryCondition::assignSysBoundary(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+                                 FsGrid< fsgrids::technical, FS_STENCIL_WIDTH> & technicalGrid) {
+
+      bool doAssign;
+      array<bool,6> isThisCellOnAFace;
+      
+      // Assign boundary flags to local DCCRG cells
+      const vector<CellID>& cells = getLocalCells();
+      for(const auto& dccrgId : cells) {
+         SpatialCell* cell = mpiGrid[dccrgId];
+         if (cell->sysBoundaryFlag == sysboundarytype::DO_NOT_COMPUTE) 
+            continue;
+         determineFace(isThisCellOnAFace, cell);
+         
+         // Comparison of the array defining which faces to use and the array telling on which faces this cell is
+         doAssign = false;
+         for (int j=0; j<6; j++) 
+            doAssign = doAssign || (facesToProcess[j] && isThisCellOnAFace[j]);
+         const auto nbrs = mpiGrid.get_face_neighbors_of(dccrgId);
+         for (uint j = 0; j < nbrs.size(); j++) {
+            CellID neighbor = nbrs[j].first;
+            if (neighbor) {
+               determineFace(isThisCellOnAFace, mpiGrid[neighbor]);
+               for (int j=0; j<6; j++) 
+                  doAssign = doAssign || (facesToProcess[j] && isThisCellOnAFace[j]);
+            }
+         }
+         if(doAssign) {
+            mpiGrid[dccrgId]->sysBoundaryFlag = this->getIndex();
+         }         
+      }
```

Comment:
Should there also be assignment of the sysBoundaryFlag onto technicalgrid here? It seems to be missing.

Reply:
This is done using feedBoundaryIntoFsGrid in gridGlue.